engine/src/uidc.cpp

```diff
@@ -1690,22 +1695,29 @@ Boolean MCUIDC::parsecolor(const MCString &s, MCColor *color, char **cname)
 	}
 	// check for a numeric second argument (Green value)
 	i2 = MCU_strtol(sptr, l, ',', done);
+	
+	if (i2 < 0 || i2 > 255)
+		return False;
+		
 	if (!done)
 	{
 		// MDW-2013-06-12: [[ Bug 10950 ]] non-numeric second argument present
 		if (l != 0)
 			return False;
 		// only a single integer as the color specification,
-		// restrict it to 0-255 and get values from sccolors array
-		i1 = MCU_max(1, MCU_min(i1, 256)) - 1;
-		i3 = sccolors[i1].blue;
-		i2 = sccolors[i1].green;
-		i1 = sccolors[i1].red;
+		// get values from sccolors array
```

Comment:
This fixes another bug:
answer color with 255 -->
Expected result: Color picker opens and white color is selected
Observed result: Color picker opens and black color is selected

The opposite happens if executing: answer color with 0